### PR TITLE
Fix setting monitorAggregationLevel to max reflects via CLI

### LINKED ISSUE #12000: Setting MonitorAggregationLevel=max via CLI when the previous value was `medium` doesn't reflect the change to `max`
## Bug report

**General Information**

- 1.7.4
- (`master`) as well

**How to reproduce the issue**

1. Deploy Cilium in any environment
2. `cilium config`
3. `cilium config MonitorAggregationLevel=max`
4. Observe that `cilium config` doesn't reflect the updated level, due to https://github.com/cilium/cilium/blob/4673688c3fd82c4ecfab582b39aa381867bd69b5/pkg/option/monitor.go#L55

This is a minor UI bug, but can be misleading to users.

### PR DESCRIPTION
**Pull request by soumynathan**
This PR fixes setting MonitorAggregationLevel to max to correctly
reflect through CLI.

Fixes: #12000
Signed-off-by: Swaminathan Vasudevan <svasudevan@suse.com>

### CODE REVIEW

File: pkg/option/monitor.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ const (
 
 	// MonitorAggregationLevelMax is the maximum level of aggregation
 	// currently supported.
-	MonitorAggregationLevelMax = MonitorAggregationLevelMedium
+	MonitorAggregationLevelMax = 4
```
Comment (aanm) [inline pkg/option/monitor.go]: `MonitorAggregationLevelMax OptionSetting = 4`

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/option/monitor.go (hunk 1 above)
Comment (soumynathan) [reply pkg/option/monitor.go]: Done.

**Review by aanm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
Unrelated to this PR: I think we could simplify the definition of the `MonitorAggregationLevel*` consts a bit by using `iota`.

**Comment by tklauser** [comment]
Marking as `needs-backport/1.7` since this was reported against 1.7.4 in #12000

**Comment by tklauser** [comment]
test-me-please

**Review by aanm — CHANGES_REQUESTED** [review]
I gave it a little mor thought on this PR, I'm not sure if setting the maximum aggregation level to 4 is the correct approach here since we only support at maximum 3, not 4 levels of aggregation. cc @joestringer are there any implications in regarding the datapath implementation i.e., user setting it to 4 while the datapath only supports 3 aggregation levels?

**Comment by christarazi** [comment]
I think the core of the issue here is that previously the `format` function for MonitorAggregationLevel didn't have a field for `max`. @soumynathan correctly fixes this. IMO, I don't think it matters whether we create a new level (in this case it is a `4`) to fix this bug, of course assuming no datapath implications. Only the formatting is affected as it seems to be a cosmetic bug, *I believe*.

Now what we are dealing with is whether we want to allow the user to specify `max` and if we should spit a warning msg to inform them that max aggregation hasn't been implemented (or similar). When I briefly synced with @joestringer on this, currently `max` is just a placeholder until we actually implement it.

**Comment by joestringer** [comment]
There's clearly a bit of confusion around what these settings are and what they mean, and that's partly on me for not documenting; Furthermore I fully anticipate that 99% of users will use whatever is the default so I haven't seen it as a tweak that's worth documenting fully at this point.

Here's the code:

https://github.com/cilium/cilium/blob/a8066cec55fca1a898db1a8d5062af6e0ce12e36/bpf/lib/trace.h#L46-L51

"NONE" on the datapath maps to the none on the daemon, that means send all datapath notifications for receive/transmit packets.
"AGGREGATE_RX" is equivalent to "low" in the daemon. That means stop sending datapath notifications for received packets. All transmitted packets should still cause a notification to be sent.
"ACTIVE_CT" (note ==`3`) adds connection aggregation which by default limits trigger notifications to once per 5 seconds per connection (unless there's new information like new TCP flags). All dropped packets still trigger notifications.

If someone is particularly careful about the semantics they need from the notifications mechanism, they can lock the value to one of the specific levels to get those semantics.

There is no "maximum" in the datapath, maximum in the daemon is a concept that says "I don't care, give me whatever notifications that the Cilium maintainers think is a reasonable level of filtering". This is part of why there's this "UI bug" right now, because "maximum" isn't a distinct value or setting with its own semantics. Its semantics *will change* if we introduce additional levels of aggregation (such as sending fewer notifications for dropped packets).

> are there any implications in regarding the datapath implementation[...]?

No, the datapath checks against ">=" specific values of aggregation:

https://github.com/cilium/cilium/blob/a8066cec55fca1a898db1a8d5062af6e0ce12e36/bpf/lib/trace.h#L121-L140

In summary, setting to 4 should have no behaviour difference and will retain the semantics we expect. We may need to bump it again in future when adding additional levels, but we can address that if/when we come to that.

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
I don't really consider this worth backporting since it's been there since Cilium v1.2 and it took until v1.8 (1.5 years+) for anyone to notice.
